pkg/operator/ceph/object/spec.go

```diff
@@ -727,22 +728,33 @@ func (c *clusterConfig) generateService(cephObjectStore *cephv1.CephObjectStore)
 	return svc
 }
 
-func (c *clusterConfig) generateEndpoint(cephObjectStore *cephv1.CephObjectStore) *discoveryv1.EndpointSlice {
+func (c *clusterConfig) generateEndpoint(cephObjectStore *cephv1.CephObjectStore) (*discoveryv1.EndpointSlice, error) {
 	labels := getLabels(cephObjectStore.Name, cephObjectStore.Namespace, true)
 
 	// Convert to string addresses
 	k8sEndpointAddrs := []string{}
 	for _, rookEndpoint := range cephObjectStore.Spec.Gateway.ExternalRgwEndpoints {
 		k8sEndpointAddrs = append(k8sEndpointAddrs, rookEndpoint.IP)
 	}
+	var addressType discoveryv1.AddressType
+	if len(k8sEndpointAddrs) > 0 {
+		ip, err := netip.ParseAddr(k8sEndpointAddrs[0])
+		if err != nil {
+			return nil, errors.Errorf("failed to get the address type for ip address %q", k8sEndpointAddrs[0])
+		}
+		addressType = discoveryv1.AddressTypeIPv4
+		if ip.Is6() {
+			addressType = discoveryv1.AddressTypeIPv6
```

Comment:
i think you can also add a validation that all the adresses in k8sEndpointAddrs or either IPv4 or IPV6 instead of assumption.